### PR TITLE
Search events with Form & Loader

### PR DESCRIPTION
**Pull request by agus-wesly**
Using Form with default get method to trigger loader. Fetch filtered events and display it. I try to use the installed react-icons library to get search icon, but it throws error. So i use svg for temporary search icon.

### CODE REVIEW

**Comment by mhaidarhanif** [comment]
@agus-wesly Use Iconify React to handle the icon

It's already setup like in the social media Icons in the footer

**Review by mhaidarhanif — CHANGES_REQUESTED** [review]
No need for useState, rely on the browser URL state when searching and filtering

When the search form submit with GET method, there will be URL search params

Use it to manage the search keyword state

**Comment by agus-wesly** [comment]
> No need for useState, rely on the browser URL state when searching and filtering
> 
> When the search form submit with GET method, there will be URL search params that can be handled in the loader
> 
> Use it to manage the search keyword state

I already remove useState and do exactly this. Am i missing something ?

File: app/components/ui/search-input.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { cn } from "~/utils/cn"
 
 export interface SearchInputProps
 	extends React.InputHTMLAttributes<HTMLElement> {
-	setValue: React.ChangeEventHandler<HTMLInputElement>
+	setValue?: React.ChangeEventHandler<HTMLInputElement>
```
Comment (mhaidarhanif) [inline app/components/ui/search-input.tsx:7]: I meant this part is not needed

File: app/components/ui/search-input.tsx [hunk 2]
```diff
@@ -13,21 +13,19 @@ export const SearchInput = React.forwardRef<HTMLInputElement, SearchInputProps>(
...
+				ref={ref}
+				type={type}
+				name={name}
+				className={cn(
+					"w-full rounded-full bg-secondary-foreground/10 px-6 py-4 text-xs font-medium outline-none backdrop-blur-sm placeholder:text-secondary-foreground/60 md:text-base",
+					className,
+				)}
+				placeholder={placeholder}
+				value={value}
+				onChange={setValue}
```
Comment (mhaidarhanif) [inline app/components/ui/search-input.tsx:26]: This part also not needed

**Review by mhaidarhanif — CHANGES_REQUESTED** [review]
I see, the useState already removed

Then need to remove the stateful related code in other parts

**Comment by mhaidarhanif** [comment]
Also, use a more common query `q` to handle the search keyword

Because after the search, there will be a time to combine it with the pagination queries

**Review by mhaidarhanif — APPROVED** [review]
Looks okay for now. Although later need to improve the searching state or remove it because there's a jank when transitioning.

**Pull request merged**